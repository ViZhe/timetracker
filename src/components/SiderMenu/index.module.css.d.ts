
export const sider: string;
export const logo: string;
export const logoIcon: string;
export const logoName: string;
