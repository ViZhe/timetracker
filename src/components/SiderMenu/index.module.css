
.sider {
  position: relative;
  z-index: 10;
  min-height: 100vh;
  box-shadow: 2px 0 6px rgba(0, 21, 41, 0.35);
}

.logo {
  overflow: hidden;
  display: flex;
  align-items: center;
  height: 32px;
  margin: 16px;
  padding-left: .5rem;
}

.logoIcon {
  font-size: 32px;
  color: #fff;
}

.logoName {
  margin-left: 1rem;
  color: #fff;
  font-size: 20px;
  transition: all .25s ease;
}

:global(.ant-layout-sider-collapsed) .logoName {
  margin-left: .5rem;
  opacity: 0;
}
