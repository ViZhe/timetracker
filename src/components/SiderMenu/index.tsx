
import { Icon, Layout, Menu } from 'antd';
import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import styles from './index.module.css';


interface IProps {
  collapsed: boolean;
  onCollapse: () => void;
}

const { Sider } = Layout;

class SiderMenu extends Component<IProps, {}> {
  render() {
    const { collapsed, onCollapse }: IProps = this.props;

    return (
      <Sider
        className={styles.sider}
        collapsible={true}
        collapsed={collapsed}
        onCollapse={onCollapse}
      >
        <Link to="/">
          <div className={styles.logo}>
            <Icon type="clock-circle" className={styles.logoIcon} />
            <span className={styles.logoName}>TimeTracker</span>
          </div>
        </Link>
        <Menu theme="dark" mode="inline" defaultSelectedKeys={['time']}>
          <Menu.Item key="time">
            <Link to="/time">
              <Icon type="home" />
              <span>Home</span>
            </Link>
          </Menu.Item>
          <Menu.Item key="reports">
            <Link to="/reports">
              <Icon type="pie-chart" />
              <span>Reports</span>
            </Link>
          </Menu.Item>
          <Menu.Item key="settings">
            <Link to="/settings">
              <Icon type="setting" />
              <span>Settings</span>
            </Link>
          </Menu.Item>
          <Menu.Item key="notmatch">
            <Link to="/notmatch">
              <Icon type="lock" />
              <span>Error</span>
            </Link>
          </Menu.Item>
        </Menu>
      </Sider>
    );
  }
}


export default SiderMenu;
