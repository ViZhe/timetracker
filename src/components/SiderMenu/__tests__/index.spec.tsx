
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import SiderMenu from '../../SiderMenu';


describe('<SiderMenu />', () => {
  let component:ShallowWrapper;
  const collapsed = false;
  const onClick = jest.fn();

  beforeEach(() => {
    component = shallow(
      <SiderMenu
        collapsed={collapsed}
        onCollapse={onClick}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
