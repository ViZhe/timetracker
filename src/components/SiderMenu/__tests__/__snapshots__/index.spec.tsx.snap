// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiderMenu /> should match snapshot 1`] = `
<Sider
  className="sider"
  collapsed={false}
  collapsedWidth={80}
  collapsible={true}
  defaultCollapsed={false}
  onCollapse={[MockFunction]}
  reverseArrow={false}
  style={Object {}}
  theme="dark"
  width={200}
>
  <Link
    replace={false}
    to="/"
  >
    <div
      className="logo"
    >
      <Icon
        className="logoIcon"
        type="clock-circle"
      />
      <span
        className="logoName"
      >
        TimeTracker
      </span>
    </div>
  </Link>
  <Menu
    className=""
    defaultSelectedKeys={
      Array [
        "time",
      ]
    }
    focusable={false}
    mode="inline"
    theme="dark"
  >
    <MenuItem
      key="time"
    >
      <Link
        replace={false}
        to="/time"
      >
        <Icon
          type="home"
        />
        <span>
          Home
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="reports"
    >
      <Link
        replace={false}
        to="/reports"
      >
        <Icon
          type="pie-chart"
        />
        <span>
          Reports
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="projects"
    >
      <Link
        replace={false}
        to="/projects"
      >
        <Icon
          type="project"
        />
        <span>
          Projects
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="settings"
    >
      <Link
        replace={false}
        to="/settings"
      >
        <Icon
          type="setting"
        />
        <span>
          Settings
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="notmatch"
    >
      <Link
        replace={false}
        to="/notmatch"
      >
        <Icon
          type="lock"
        />
        <span>
          Error
        </span>
      </Link>
    </MenuItem>
  </Menu>
</Sider>
`;
